django/template/loader_tags.py

```diff
@@ -249,6 +250,34 @@ def do_block(parser, token):
     return BlockNode(block_name, nodelist)
 
 
+def construct_relative_path(name, relative_name):
+
+    if not relative_name.startswith('"./'):
+        # argument is variable or literal, that not contain relative path
+        return relative_name
+
+    new_name = posixpath.normpath(
+        posixpath.join(
+            posixpath.dirname(name.lstrip('/')),
+            relative_name.strip('"')
+        )
+    )
+
+    if new_name.startswith('../'):
+        raise TemplateSyntaxError(
+            "Relative name '%s' have more parent folders, then given template name '%s'"
```

Comment:
have -> has
then -> then the template name %s in which the <extends/include> tag appears